app/main.py

```diff
@@ -1 +1,21 @@
-# write your code here
+from app.cafe import Cafe
+from app.errors import (
+    NotWearingMaskError,
+    VaccineError
+)
+
+
+def go_to_cafe(friends: list[dict], cafe: "Cafe") -> str | Exception:
+    masks_to_buy = 0
+    try:
+        for friend in friends:
+            try:
+                cafe.visit_cafe(friend)
+            except NotWearingMaskError:
+                masks_to_buy += 1
+    except VaccineError:
+        return "All friends should be vaccinated"
```

Comment:
Catching `VaccineError` here might prevent specific exceptions like `NotVaccinatedError` or `OutdatedVaccineError` from being handled individually. Consider handling specific exceptions first before catching the base class to ensure more precise error handling.